### ISSUE TITLE
1.4.6 multipath memory leak

### ISSUE THREAD

**Issue by wizzard0** [zerotier/ZeroTierOne#1274]
- What you expect to be happening.
  - ZT working reliably over weeks (until machine reboots due to updates etc)
- What is actually happening?
  - Memory usage of ZT service (windows, linux) steadily grows at about 200-500MB per hour when multipath is enabled
- Any steps to reproduce the error.
  - Install ZT 1.4.6 on a number of machines having different (3+) connectivity options each

In my case: 
- laptop and desktop can reach each other via home LAN;
- VMs on laptop\desktop can reach each other via bridged OpenVPN
- My office desktop and work servers are reachable from laptop and desktop via office VPN (daemon running on each machine separately, not on router)
- office machines can reach each other via LAN, and they have multiple LAN connections each
- Office machines and home machines can reach each other over the internet (via NATs on each side)
- **Connectivity is not static**: VPNs sometimes get turned on or off (or VPN servers fail); home router fails over to different ISPs, laptop travels with me :)
- and there are a couple of my personal cloud servers too, configured as **moons** to reduce latency (really helps btw! but the memory leak occurs without moons too)

In this (albeit complex) configuration) any machine where I tried to enable multipath slowly leaks memory, regardless of OS (Win10, Server 2016, Server 2019, Server 2008R2, Ubuntu 18, Ubuntu 20, Centos 7, Centos 8, Debian 8), both on physical and virtual machines.

Also using ZT on iOS but don't know how to configure multipath there :)

Currently solved it by scheduling a service restart every 8 hours on each machine, but this is really inconvenient because sometimes it restarts in the middle of your SSH session or copying a file.

## Additional information 

**Desktop (please complete the following information):**
 - OS: **Looks OS-independent**
 - OS/Distribution Version: 
Win10 v1908, v2004; Server 2016, Server 2019, Server 2008R2, Ubuntu 18, Ubuntu 20, Centos 7, Centos 8, Debian 8
 - ZeroTier Version: **1.4.6**
  
**Additional context**
- ZeroTier Network Configuration: IPv4 & IPv6 networks defined on your ZeroTier Central
  - All machines are joined into a **single network** with /24 ipv4 DHCP and some static IPs. 
  - **Default flow rules**

- Router Config: are you permitting port 9993, uPnP, and NAT-PMP?
- Firewall Config: are you permitting port 9993 on your OS; setting it to "Private" on Windows?
  - Toggling 9993, upnp, nat-pmp on-off, firewall configuration **has no effect** on the memory leak.
- Are you using this at home, in an office, college, etc?
  - Linking machines I use daily to combat multiple flaky networks and routing misconfigurations into a single /24 subnet, for convenience.

**Comment by glimberg** [comment]
You're using the old beta multipath on 1.4.6?  That was beta and is being superseded in 1.6.0 by a new implementation which we have the 1.5.0 beta release out now and available here:

https://download.zerotier.com/PRERELEASES/1.5.0/dist/

Note: Windows isn't code signed yet.  Waiting on a new signing key.

**Comment by joseph-henry** [comment]
@wizzard0 Also please check out this documentation for `1.5.0+` multipath: https://zerotier.atlassian.net/wiki/spaces/SD/pages/568459265/Multipath

Going to close this ticket soon since it's regarding a superseded version of this feature. Feel free to open another ticket if you have issues with `1.5.0` multipath and we can work on it there.
